scripts/getting_started.sh

```diff
@@ -0,0 +1,583 @@
+        fi
+        USER=`whoami`
+        # Add your user to the docker group.
+        runCmd "sudo usermod -aG docker ${USER}" 0 "Add ${USER} to docker group"
+        echo "You would need to log out and log back in so that your group membership is re-evaluated. Rerun this script after that."
+        exit
+    fi
+    echo "current user has permission for docker"
+}
+
+function installKind() {
+
+    cmdName="kind"
+     ## check  if denpency is installed before
+    isCmdInstalled "${cmdName}"
+
+    ## install denpency if it not installed
+    if [ $? -eq 0 ] ; then
+        echo "Installing ${cmdName}..."
+        curl -Lo ./kind https://kind.sigs.k8s.io/dl/v0.11.1/kind-linux-amd64 && sudo install kind /usr/local/bin/kind
+        ## check  if denpency is installed successfully
+        isCmdInstalled  "${cmdName}"
+        exitIfNot $? 0 "Installing ${cmdName} failed, exit..."
+    fi
+}
+
+function installClusterctl() {
+
+    cmdName="clusterctl"
+     ## check  if denpency is installed before
+    isCmdInstalled "${cmdName}"
+
+    ## install denpency if it not installed
+    if [ $? -eq 0 ] ; then
+        echo "Installing ${cmdName}..."
+        curl -L https://github.com/kubernetes-sigs/cluster-api/releases/download/v1.0.0/clusterctl-linux-amd64 -o clusterctl && sudo install clusterctl /usr/local/bin/clusterctl
+        ## check  if denpency is installed successfully
+        isCmdInstalled  "${cmdName}"
+        exitIfNot $? 0 "Installing ${cmdName} failed, exit..."
+    fi
+}
+
+
+function intallDependencies(){
+    installDocker
+    enableDocker
+    installKind
+    installClusterctl
+    commonInstall jq "sudo apt install -y jq"
+    commonInstall kubectl "sudo snap install kubectl --classic"
+    commonInstall make "sudo apt install -y build-essential"
+    commonInstall go "sudo snap install go --classic"
+}
+
+function commonInstall(){
+    cmdName=$1
+    installCmd=$2
+
+    ## check  if denpency is installed before
+    isCmdInstalled "${cmdName}"
+
+    ## install denpency if it not installed
+    if [ $? -eq 0 ] ; then
+        runCmd "${installCmd}" 0 "Installing ${cmdName}..."
+
+        ## check  if denpency is installed successfully
+        isCmdInstalled  "${cmdName}"
+        exitIfNot $? 0 "Installing ${cmdName} failed, exit..."
+    fi
+}
+
+function createKindCluster(){
+    echo  "Creating kind cluster..."
+
+    msg=$(kind create cluster 2>&1)
+    if [ $? -ne 0 ]; then
+        echo "Create kind cluster failed"
+        echo $msg | grep -q "You have reached your pull rate limit"
+        if [ $? -eq 0 ]; then 
+            echo "Suggestion: you can use \"docker login\" to avoid such an error."
+            exit
+        fi
+    else
+        echo "Create kind cluster successfully"
+    fi
+}
+
+function downloadByohCode(){
+    runCmd "rm -rf cluster-api-provider-bringyourownhost" 0 " Cleaning byoh code..."
+    msg=$(git clone git@github.com:vmware-tanzu/cluster-api-provider-bringyourownhost.git 2>&1)
```

Comment:
Why not just print the `git clone` output? We will not have to grep for error messages

Reply:
That's because sometimes it doesn't have  correct access rights to code repos. I need to catch this error and print suggestion.